### PR TITLE
Add new endpoint get(find) all filtred news

### PR DESCRIPTION
**Pull request by MariyaSin**
Task http://outsourcing.nat.tepkom.ru:20080/redmine/issues/165
1 added endpoint
2 added tests
3 renamed Db column "AuthorName" to  "Pseudonym"

### CODE REVIEW

**Review by 24eeeee — APPROVED** [review]

File: LT.DigitalOffice.NewsService.sln [hunk 1]
```diff
@@ -12,6 +12,9 @@ EndProject
 Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "NewsService.Business", "src\NewsService.Business\NewsService.Business.csproj", "{809396D8-723D-433B-9392-7BAF925C5EB5}"
 EndProject
 Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "NewsService.Data", "src\NewsService.Data\NewsService.Data.csproj", "{6084FDBE-4A64-4284-A72A-DB205C2F62EF}"
+	ProjectSection(ProjectDependencies) = postProject
+		{EA07A7B3-2039-401E-9DE0-85BC79894201} = {EA07A7B3-2039-401E-9DE0-85BC79894201}
```
Comment (BelehovEgor) [inline LT.DigitalOffice.NewsService.sln:16]: What is this?

File: src/NewsService.Data/NewsRepository.cs [hunk 2]
```diff
@@ -40,6 +42,45 @@ public Guid CreateNews(DbNews news)
             return news.Id;
         }
 
+        public List<DbNews> FindNews(FindNewsParams findNewsParams)
+        {
+            if (findNewsParams == null)
+            {
+                throw new Exception("search parameters not passed.");
```
Comment (BelehovEgor) [inline src/NewsService.Data/NewsRepository.cs:49]: write a more precise exception type

File: src/NewsService.Data/NewsRepository.cs [hunk 3]
```diff
@@ -40,6 +42,45 @@ public Guid CreateNews(DbNews news)
...
+            if (findNewsParams.Subject != null)
+            {
+                dbNewsList = dbNewsList.Where(x => x.Subject == findNewsParams.Subject);
+            }
+
+            var dbNewsResponse = dbNewsList.ToList();
+
+            if (dbNewsResponse == null)
+            {
+                throw new Exception("News was not found.");
```
Comment (BelehovEgor) [inline src/NewsService.Data/NewsRepository.cs:78]: maybe we have to return an empty list. Why do we throw exc, if all work well

File: src/NewsService.Models.Db/DbNews.cs [hunk 4]
```diff
@@ -11,14 +11,16 @@ public class DbNews
         public string Content { get; set; }
```
Comment (BelehovEgor) [inline src/NewsService.Models.Db/DbNews.cs:11]: Do you want to delete attribute and create a configuration class?

File: src/NewsService.Models.Dto/Models/FindNewsParams.cs [hunk 5]
```diff
@@ -0,0 +1,12 @@
+﻿using System;
+
+namespace LT.DigitalOffice.NewsService.Models.Dto.Models
+{
+    public class FindNewsParams
+    {
+        public Guid? AuthorId { get; set; }
+        public Guid? DepartmentId { get; set; }
+        public string Pseudonym { get; set; }
```
Comment (BelehovEgor) [inline src/NewsService.Models.Dto/Models/FindNewsParams.cs:9]: Why do you rename this field from AuthorName -> Pseudonym? for convenience?

File: test/NewsService.Business.UnitTests/FindNewsCommandTests.cs [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+        [Test]
+        public void CommandTest()
+        {
+            SerializerAssert.AreEqual(_newsResponse, _command.Execute(_goodFinedNewsParams));
+        }
+
+        [Test]
+        public void MapperNullDbNewsTest()
+        {
+            Assert.Throws<Exception>(() => _command.Execute(_finedNewsParamsReturnsNullOfDbNews));
```
Comment (BelehovEgor) [inline test/NewsService.Business.UnitTests/FindNewsCommandTests.cs:78]: there are default exceptions everywhere, is that good?

**Review by BelehovEgor — CHANGES_REQUESTED** [review]

File: src/NewsService.Models.Db/DbNews.cs (hunk 4 above)
Comment (MariyaSin) [reply src/NewsService.Models.Db/DbNews.cs:11]: that not in my task

File: src/NewsService.Models.Dto/Models/FindNewsParams.cs (hunk 5 above)
Comment (MariyaSin) [reply src/NewsService.Models.Dto/Models/FindNewsParams.cs:9]: in this property it is necessary to save not the name but the pseudonym, the author's name is obtained from the userSevravise.

File: LT.DigitalOffice.NewsService.sln (hunk 1 above)
Comment (MariyaSin) [reply LT.DigitalOffice.NewsService.sln]: removed

File: src/NewsService.Data/NewsRepository.cs (hunk 3 above)
Comment (MariyaSin) [reply src/NewsService.Data/NewsRepository.cs]: no verification is needed here

File: test/NewsService.Business.UnitTests/FindNewsCommandTests.cs (hunk 6 above)
Comment (MariyaSin) [reply test/NewsService.Business.UnitTests/FindNewsCommandTests.cs:78]: all exceptions inherit from the Exception type. I think it's better that way. the Exception will catch any type of exception thrown in the mapper

File: src/NewsService.Data/NewsRepository.cs (hunk 2 above)
Comment (MariyaSin) [reply src/NewsService.Data/NewsRepository.cs]: check removed as unnecessary

**Review by BelehovEgor — APPROVED** [review]

**Pull request merged**